### PR TITLE
modify loader flag for patients count

### LINKED ISSUE #8030: Add a skeleton loader during advance filters are loaded
**Describe the bug**

When we open a patient tab, which is already having a cache advanced filter applied to that page, for a second or two, the count of the patient will be shown as 0

Similarly, this behavior issue is happening across the platform

**To Reproduce**

**Expected behavior**

On discussion, we need to check it in packet loss 

**Screenshots**

image

### PR DESCRIPTION
**Pull request by FrankenDeba**
## Proposed Changes

- Fixes #8030 
- Added the loader instead of showing 0 on patients count

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Comment by FrankenDeba** [comment]
Screenshot 2024-06-17 at 3 40 51 PM

File: src/Components/Patient/ManagePatients.tsx [hunk 1]
```diff
@@ -902,7 +902,7 @@ export const PatientManager = () => {
             <CountBlock
               text="Total Patients"
               count={data?.count || 0}
-              loading={isLoading}
+              loading={(!data?.count && data?.count !== 0) || isLoading}
```
Comment (khavinshankar) [inline src/Components/Patient/ManagePatients.tsx:905]: @FrankenDeba make this change at CountBlock so that it is applied throughout the application

Comment (khavinshankar) [inline src/Components/Patient/ManagePatients.tsx]: @FrankenDeba make this change at CountBlock so that it is applied throughout the application

**Review by khavinshankar — CHANGES_REQUESTED** [review]

File: src/Components/Patient/ManagePatients.tsx (hunk 1 above)
Comment (FrankenDeba) [reply src/Components/Patient/ManagePatients.tsx]: @khavinshankar Done, please check it!

File: src/CAREUI/display/Count.tsx [hunk 2]
```diff
@@ -22,7 +22,7 @@ export default function CountBlock(props: Props) {
           <dt className="my-2 truncate text-sm font-semibold text-gray-700">
             {props.text}
           </dt>
-          {props.loading ? (
+          {props.loading || (!props.count && props.count !== 0) ? (
```
Comment (sainak) [inline src/CAREUI/display/Count.tsx:25]: wouldn't this be stuck in the loading state if the count is actually 0?

we should not collace count to 0 if it's undefined while passing it to the component, and hide loading state when the count becomes defined

Comment (FrankenDeba) [reply src/CAREUI/display/Count.tsx:25]: Didn't understand,I am already checking if the count is not 0,do I still need to change the approach?
@sainak

Comment (FrankenDeba) [reply src/CAREUI/display/Count.tsx:25]: You there?
Can you please be more specific about the changes?
@sainak

**Pull request closed without merging**